azkaban-web-server/src/main/java/azkaban/imagemgmt/services/ImageRampupServiceImpl.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.services;
+
+import static azkaban.Constants.ImageMgmtConstants.IMAGE_TYPE;
+
+import azkaban.imagemgmt.daos.ImageRampupDao;
+import azkaban.imagemgmt.dto.ImageMetadataRequest;
+import azkaban.imagemgmt.exeception.ImageMgmtException;
+import azkaban.imagemgmt.exeception.ImageMgmtValidationException;
+import azkaban.imagemgmt.models.ImageRampupPlan;
+import azkaban.imagemgmt.models.ImageRampupPlanRequest;
+import azkaban.imagemgmt.models.ImageRampupRequest;
+import azkaban.imagemgmt.models.ImageRampupRequest.StabilityTag;
+import azkaban.imagemgmt.utils.ConverterUtils;
+import azkaban.imagemgmt.utils.ValidatorUtils;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This service layer implementation for processing and delegation of image rampup APIs. For example
+ * API request processing and validation are handled in this layer. Eventually the requests are
+ * routed to the DAO layer for data access.
+ */
+@Singleton
+public class ImageRampupServiceImpl implements ImageRampupService {
+
+  private static final Logger log = LoggerFactory.getLogger(ImageRampupServiceImpl.class);
+
+  private final ImageRampupDao imageRampupDao;
+  private final ConverterUtils converterUtils;
+
+  @Inject
+  public ImageRampupServiceImpl(final ImageRampupDao imageRampupDao,
+      final ConverterUtils converterUtils) {
+    this.imageRampupDao = imageRampupDao;
+    this.converterUtils = converterUtils;
+  }
+
+  @Override
+  public int createImageRampupPlan(ImageMetadataRequest imageMetadataRequest)
+      throws IOException, ImageMgmtException {
+    // Convert input json payload to ImageRampupPlanRequest object
+    ImageRampupPlanRequest imageRampupPlanRequest = converterUtils
```

Comment:
Who can create ramp up plan? Where is that validation?

Reply:
Azkaban developer or image type owner can create rampup plan. Validation related permission will be added as part of API permission and access task.